src/lib.rs

```diff
@@ -206,9 +206,9 @@ impl <T : PartialEq + HasName + ::std::fmt::Debug + ::std::clone::Clone,
             false
         } else if self.nodes.len() < optimal_table_size() {
             true
-        } else  if ::xor_name::closer_to_target(their_name,
-                                                self.nodes[self.dynamic_group_size()].name(),
-                                                &self.our_name) {
+        } else if ::xor_name::closer_to_target(their_name,
+                                               self.furthest_close_node().unwrap().name(),
```

Comment:
Not so sure about the unwrap here. We tend to not use ignore or expect and here I would return an error ? perhaps.